### PR TITLE
List Invites endpoint

### PR DESCRIPTION
**Pull request by talavis**
> **Before submitting the PR, please go through the sections below and fill in what you can. If there are any items that are irrelevant for the current PR, remove the row. If a relevant option is missing, please add it as an item and add a PR comment informing that the new option should be included into this template.**

> **All _relevant_ items should be ticked before the PR is merged**

# Description

Adds an endpoint to the API for listing invites at `/user/invites`.

* All users get a response with `created`, `email`, `role`, `projects`
* Researchers can only list invites for projects where they are the owner
* Unit Admin|Personnel can list invites for the unit and any projects connected to the unit
* Superadmin can list all invites
  - Will also include a `unit` property 

Implements Jira Issue DDS-1353.

- [X] Add a summary of the changes and the related issue
- [X] Fixes [link to issue / Jira issue ID]

## Type of change

- [X] New feature (non-breaking)

# Checklist:

## General

- [x] [Changelog](../CHANGELOG.md): New row added. Not needed when PR includes _only_ tests.
- [X] Code change
  - [X] Self-review of code done
  - [x] Comments added, particularly in hard-to-understand areas

## Repository / Releases

- [X] Rebase / update of branch done

## Checks

- [x] Formatting: Black & Prettier checks pass
- [x] CodeQL passes
- [x] Tests
  - [x] I have added tests for the new code
  - [x] The tests pass
- [x] Trivy / Snyk:
  - [x] There are no new security alerts

<a href="https://gitpod.io/#https://github.com/ScilifelabDataCentre/dds_web/pull/1294"></a>

### CODE REVIEW

File: dds_web/api/user.py
Comment (inaod568) [inline dds_web/api/user.py:1271]: Slight simplification
```suggestion
            unit = current_user.unit
            unit_projects = set(proj.id for proj in unit.projects)
```

File: dds_web/api/user.py
Comment (inaod568) [inline dds_web/api/user.py:1264]: Not used I think?

File: dds_web/api/user.py
Comment (inaod568) [inline dds_web/api/user.py:1274]: Add a little comment above this to explain for those that just want to skim and maybe understand less of the code

File: dds_web/api/user.py
Comment (inaod568) [inline dds_web/api/user.py:1298]: Short comment above this next part too

File: dds_web/api/user.py
Comment (inaod568) [inline dds_web/api/user.py:1313]: Comment about what's done here

**Review by inaod568 — CHANGES_REQUESTED** [review]
I'd like to add a check at the beginning of the endpoint too that raises an exception if the user is not a project owner in any ongoing projects. I think users will be confused otherwise regarding why they can't see any invites.

**Review by inaod568 — CHANGES_REQUESTED** [review]
I like the `row_to_dict` function. We need to add a little change though in order to clarify some of the info but otherwise 👍🏻 Also after this we need a test or two to verify

File: dds_web/api/user.py
Comment (talavis) [reply dds_web/api/user.py:1292]: It is added if the `Unit Admin` is adding the new `Unit *`. I think it makes sense to have projects shown as it immediately shows if more work will be needed once the person has been added to the unit.

Comment (talavis) [reply dds_web/api/user.py:1292]: It can be skipped for superadmins, but I don't think it's a problem

File: dds_web/api/user.py
Comment (talavis) [reply dds_web/api/user.py:1313]: This is not true if a new project is created while the invite is active, right?

File: dds_web/api/user.py
Comment (inaod568) [reply dds_web/api/user.py:1292]: Never mind @talavis

File: dds_web/api/user.py
Comment (inaod568) [reply dds_web/api/user.py:1313]: Yes ignore this. https://github.com/ScilifelabDataCentre/dds_web/pull/1294#discussion_r991024561

**Comment by inaod568** [comment]
@talavis Can you rebase?

**Review by inaod568 — APPROVED** [review]
Looks good! Waiting for @valyo before merge.

**Review by valyo — APPROVED** [review]
Looks good and works as expected

**Pull request merged**
